### PR TITLE
Fix null sender id in Synthetic Nft Allowance Migration

### LINKED ISSUE #6150: Synthetic Nft Allowance Owner Migration fails for null owner
### Description

The following error occurred:

```
ERROR: null value in column "owner" of relation "nft_allowance_temp" violates not-null constraint\\n  Detail: Failing row contains (t, 1655500871514259667, null, 968621, 1007208, 914500).\\n	at org.springframework.jdbc.support.SQLStateSQLExceptionTranslator.doTranslate(SQLStateSQLExceptionTranslator.java:105)\\n
```

### Steps to reproduce

1. Run the migration on mainnet.

### Additional context

_No response_

### Hedera network

mainnet

### Version

v0.82.0-rc1 

### Operating system

None

### PR DESCRIPTION
**Pull request by edwin-greene**
**Description**:
Fix attempting to insert a null sender id as a non-null owner.

**Related issue(s)**:

Fixes #6150 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by xin-hedera — CHANGES_REQUESTED** [review]

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/SyntheticNftAllowanceOwnerMigration.java [hunk 1]
```diff
@@ -48,7 +48,8 @@ with affected as (
                 union all
                 select * from nft_allowance_history
               ) nfta
-              join contract_result cr on cr.consensus_timestamp = lower(nfta.timestamp_range)
+              join contract_result cr on cr.sender_id is not null and
```
[the remark this answers is not in the document]
Comment (edwin-greene) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/SyntheticNftAllowanceOwnerMigration.java]: Thanks I have moved it to the where clause for `delete_nft_allowance` and `delete_nft_allowance_history`.

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/SyntheticNftAllowanceOwnerMigration.java [hunk 2]
```diff
@@ -52,7 +52,7 @@ with affected as (
             ), delete_nft_allowance as (
               delete from nft_allowance nfta
               using affected a
-              where nfta.owner in (a.owner, a.sender_id) and nfta.spender = a.spender and nfta.token_id = a.token_id
+              where nfta.owner in (a.owner, a.sender_id) and a.sender_id is not null and nfta.spender = a.spender and nfta.token_id = a.token_id
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/SyntheticNftAllowanceOwnerMigration.java:55]: The `a.sender_id is not null` should be in the where clause of the `with affected` CTE.

**Review by steven-sheehy — APPROVED** [review]
LGTM

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/SyntheticNftAllowanceOwnerMigration.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (edwin-greene) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/SyntheticNftAllowanceOwnerMigration.java]: Scratch that. I have simplified it to the `where` clause of the CTE per Steven's suggestion.

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**